Parse each colon separated expr as an assignement

hochata opened the pull request:
Before, the macro iterated each expression children and guessed what to do using their kind.

Now, each expression is directly divided as a name and a value to create the assignment.

This way, identifiers can be used on the right hand side of the assignment.

Solves #3 and would make #2 trivial to solve

Commit:
Parse each colon separated expr as an assignement

Before, the macro iterated each expression children and guessed what
to do using their kind.

Now, each expression is directly divided as a name and a value to
create the assignement.

This way, identifiers can be used on the right hand side of the assignement

juancarlospaco commented on src/cliche.nim:
```diff
@@ -79,52 +79,40 @@ macro getOpt*(source: seq[string]; variables: untyped; helpMessage: static[strin
   for key_value in variables:
-    for item in key_value:
-      if item.kind == nnkIdent:
-        for c in item.strVal:
-          doAssert c in {'a'..'z', '0'..'9', '_'}, "Name must be ASCII Lowercase"
```
But this for loop checks that the name is a valid name,
because valid names in Nim may not be valid in the terminal,
this check of the chars of the name should be always enforced to avoid bugs.

Example:
```Nim
let `foo   bar` = 42
```

is valid Nim but will fail on terminal because of whitespaces etc.
To be safe I prefer it to use ASCII variable names only, with no whitespaces.
We can not merge without that check.

hochata replied:
Oh, yeah. I didn't think about that. I will restore the check.

Commit:
Restoring identifiers checks

Now using `validIdentifier` to check names. That is, an ASCII letter
followed by more ASCII letters and numbers.

I delete the original check because I didn't think it was useful, but I
forgot stromping exists and you can get really wacky named identifiers

juancarlospaco commented:
Theres still conflicting files, it wont allow me to Merge.

hochata committed:
Merge branch 'nim' into separate-idents

hochata commented:
Ready to merge
